tests/integration/testsuites/gateway/features/gateway.feature

```diff
@@ -0,0 +1,12 @@
+Feature: Checking default kyma gateway
+  Scenario: Kyma gateway is deployed
+    Given there is an APIGateway operator in "Ready" state
+    Then there is a "kyma-gateway" gateway in "kyma-system" namespace
+    And there is a "kyma-gateway-certs" secret in "istio-system" namespace
+
+  Scenario: Kyma gateway cannot be disabled when there is an APIRule
```

Comment:
The scenario describes a simple scenario, when the actual test does way more.
Also the test doesn't fully verify, that kyma-gateway is not removed. It verifies that the CR is in warning state:
`Then disabling kyma gateway will result in "Warning" due to existing APIRule`, but the kyma-gateway could still be removed from the cluster.
I'd also separate this tests to have a clear scenario that is tested.